scripts/reset-simulators.sh

```diff
@@ -1,59 +1,38 @@
 #!/bin/bash
 
-set -o pipefail
-set -e
+source "$(dirname "$0")/swift-version.sh"
 
 while pgrep -q Simulator; do
     # Kill all the current simulator processes as they may be from a
     # different Xcode version
-    pkill Simulator 2>/dev/null || true
+    pkill Simulator 2>/dev/null
     # CoreSimulatorService doesn't exit when sent SIGTERM
-    pkill -9 Simulator 2>/dev/null || true
-  done
-
-# Shut down simulators until there's no booted ones left
-# Only do one at a time because devices sometimes show up multiple times
-while xcrun simctl list | grep -q Booted; do
-  xcrun simctl list | grep Booted | sed 's/.* (\(.*\)) (Booted)/\1/' | head -n 1 | xargs xcrun simctl shutdown
+    pkill -9 Simulator 2>/dev/null
 done
 
-# Clean up all available simulators
-(
-    previous_device=''
-    IFS=$'\n' # make newlines the only separator
-    for LINE in $(xcrun simctl list); do
-        if [[ $LINE =~ unavailable || $LINE =~ disconnected ]]; then
-            # skip unavailable simulators
-            continue
-        fi
-
-        if [[ $LINE =~ "--" ]]; then
-            # Reset the last seen device so we won't consider devices with the same name to be duplicates
-            # if they appear in different sections.
-            previous_device=""
-            continue
-        fi
-
-        regex='^(.*) [(]([0-9A-F]{8}-[0-9A-F]{4}-[0-9A-F]{4}-[0-9A-F]{4}-[0-9A-F]{12})[)]'
-        if [[ $LINE =~ $regex ]]; then
-            device="${BASH_REMATCH[1]}"
-            guid="${BASH_REMATCH[2]}"
+# Shut down booted simulators
+xcrun simctl list devices -j | jq -c -r '.devices | flatten | map(select(.availability == "(available)")) | map(select(.state == "Booted")) | map(.udid) | .[]' | while read udid; do
```

Comment:
Does this do something different than the old code, or is it just a more concise way of achieving the same thing? Relying on a third-party tool being installed for this script to work makes me a little sad.